ipl/main.c

```diff
@@ -1940,6 +1937,13 @@ int fix_attributes(char *path, u32 *total)
 			memcpy(&path[i+1], fno.fname, k + 1);
 			path[i + k + 2] = 0;
 
+			if (strcmp(path, "//Nintendo") == 0)
```

Comment:
That's crazy that the timings have to be so tight. I love learning this kind of stuff!

Reply:
They don't need to be too tight here, but it's good to save some ms * no of files.

Reply:
In that case I think I can make this even faster by not checking unless the currently opened directory is root.

Reply:
Indeed. You can move it upwards after 
```
// Set new directory.
i = strlen(path);
```
and do 

`if (!strcmp(path, "//") && (!strcmp(fno.fname, "Nintendo"))`

I don't remember how armv4t behaves on that though.
Normally it should break on the first check being false, because next we have `&&`

Reply:
I'm thinking even more efficient than that. You're right that it does short-circuit if the first one is false. If we make the first one a boolean check, it'll be much faster than a strcmp. You can see in my followup commit.